### PR TITLE
fix(core-common): fix missing env at StorybookConfig

### LINKED ISSUE #21733: [Bug] 'env' does not exist in type 'StorybookConfig'
### Describe the bug

``` ts
// main.ts

import type { StorybookConfig } from '@storybook/nextjs'

const config: StorybookConfig = {
  stories: ['../src/**/*.mdx', '../src/**/*.stories.@(js|jsx|ts|tsx)'],
  staticDirs: ['../public'],
  addons: [
    'storybook-dark-mode',
    '@storybook/addon-links',
    '@storybook/addon-essentials',
    '@storybook/addon-interactions',
  ],
  core: {
    disableTelemetry: true,
  },
  framework: {
    name: '@storybook/nextjs',
    options: {
      strictMode: true,
    },
  },
  docs: {
    // see below for alternatives
    defaultName: 'Docs',
    // set to change the name of generated docs entries
    autodocs: true,
  },
// error happened at here
  env: (config: any) => ({
    ...config,
    NEXT_PUBLIC_WCS_BE_API_HOST_URL: 'https://example.com/',
    // fake https for chromatic deployment
    NEXT_PUBLIC_QIII_SOCKET_URL: 'https://socket.example.com',
  }),
}
module.exports = config
```

show error: 

```ts
Type '{ stories: string[]; staticDirs: string[]; addons: string[]; core: { disableTelemetry: true; }; framework: { name: "@storybook/nextjs"; options: { strictMode: true; }; }; docs: { defaultName: string; autodocs: true; }; env: (config: any) => any; }' is not assignable to type 'StorybookConfig'.
  Object literal may only specify known properties, and 'env' does not exist in type 'StorybookConfig'.
```

### To Reproduce

_No response_

### System

```shell
System:
    OS: macOS 13.2.1
    CPU: (8) arm64 Apple M1
  Binaries:
    Node: 16.14.0 - ~/.nvm/versions/node/v16.14.0/bin/node
    npm: 8.3.1 - ~/.nvm/versions/node/v16.14.0/bin/npm
  Browsers:
    Chrome: 111.0.5563.64
    Safari: 16.3
  npmPackages:
    @storybook/addon-actions: ^7.0.0-rc.5 => 7.0.0-rc.5 
    @storybook/addon-essentials: ^7.0.0-rc.5 => 7.0.0-rc.5 
    @storybook/addon-interactions: ^7.0.0-rc.5 => 7.0.0-rc.5 
    @storybook/addon-links: ^7.0.0-rc.5 => 7.0.0-rc.5 
    @storybook/csf: 0.0.2-next.10 => 0.0.2-next.10 
    @storybook/nextjs: ^7.0.0-rc.5 => 7.0.0-rc.5 
    @storybook/react: ^7.0.0-rc.5 => 7.0.0-rc.5 
    @storybook/testing-library: ^0.0.14-next.1 => 0.0.14-next.1 
    @storybook/theming: ^7.0.0-rc.5 => 7.0.0-rc.5
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by zigang93**
Closes #21733 

## What I did

add missing env at StorybookConfig from error:

```ts
Type '{ stories: string[]; staticDirs: string[]; addons: string[]; core: { disableTelemetry: true; }; framework: { name: "@storybook/nextjs"; options: { strictMode: true; }; }; docs: { defaultName: string; autodocs: true; }; env: (config: any) => any; }' is not assignable to type 'StorybookConfig'.
  Object literal may only specify known properties, and 'env' does not exist in type 'StorybookConfig'.
```

## How to test

- 

## Checklist

- [ ] Make sure your changes are tested (stories and/or unit, integration, or end-to-end tests)
- [ ] Make sure to add/update documentation regarding your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

- [ ] If this PR should be tested against many or all sandboxes,
      make sure to add the `ci:merged` or `ci:daily` GH label to it.
- [ ] Make sure this PR contains **one** of the labels below.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

### CODE REVIEW

**Comment by IanVS** [comment]
Thanks @zigang93! It looks like I put it in the wrong spot in https://github.com/storybookjs/storybook/pull/21173.  Would you mind also removing the type from `BuilderOptions` (where I had put it) now?

**Review by IanVS — APPROVED** [review]
LGTM, but I'm the one who got it wrong before so take that with a grain of salt.  🙈

**Comment by kasperpeulen** [comment]
Maybe @ndelangen can have another look?

File: code/lib/types/src/modules/core-common.ts [hunk 1]
```diff
@@ -356,7 +355,10 @@ export interface StorybookConfig {
     config: TransformOptions,
     options: Options
   ) => TransformOptions | Promise<TransformOptions>;
-
+  /**
+   * Modify or return env config.
+   */
+  env?: (envs: Record<string, string>) => Record<string, string>;
```
Comment (ndelangen) [inline code/lib/types/src/modules/core-common.ts:361]:
```suggestion
  env?: PresetValue<Record<string, string>>;
```

**Review by ndelangen — APPROVED** [review]

**Review by ndelangen — APPROVED** [review]

**Pull request merged**